### PR TITLE
[BUILD] Add a CMake option to disable shared libs

### PR DESCRIPTION
**Pull request by tobim**
This option is useful to support platforms that do not support dynamic libraries and `dlopen`.

Fixes # (issue)

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by tobim** [comment]
It is finished. The option only makes sense to use without `-DBUILD_SHARED_LIBS=ON`. Sorry for not making that clear.

File: CMakeLists.txt [hunk 1]
```diff
@@ -287,6 +287,10 @@ option(WITH_ASYNC_EXPORT_PREVIEW "Whether to enable async export" OFF)
 option(WITH_METRICS_EXEMPLAR_PREVIEW
        "Whether to enable exemplar within metrics" OFF)
 
+option(OPENTELEMETRY_STATIC_ONLY
```
Comment (lalitb) [inline CMakeLists.txt:290]: should we name it to (say) - `OPENTELEMETRY_SKIP_DYNAMIC_LOADING_TESTS` - if this is to skip generating libraries for specific tests. Or any other name, OPENTELEMETRY_STATIC_ONLY is too generic.

File: examples/plugin/plugin/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
+if(OPENTELEMETRY_SKIP_DYNAMIC_LOADING_TESTS)
```
Comment (ThomsonTan) [inline examples/plugin/plugin/CMakeLists.txt:4]: Probably better to check this in the parent CMakeList.txt, around `add_subdirectory(plugin)`?

Comment (tobim) [reply examples/plugin/plugin/CMakeLists.txt]: I updated the PR to do it this way.

**Review by marcalff — APPROVED** [review]
LGTM, thanks for the fix.

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**